downsample_cells.R

```diff
@@ -250,5 +250,7 @@ write10xCounts(opt$output_dir, assays(sce)[[1]], barcodes = sce$Barcode,
 # rename cell id field and remove redundant column
 colnames(colData(sce))[1] <- opt$cell_id_field
 colData(sce) <- colData(sce)[, -2] 
-write.table(colData(sce), opt$metadata_upd, sep="\t")
+# write outputs and avoid column names mangling
+write(paste(colnames(colData(sce)), collapse="\t"), opt$metadata_upd, sep="\t")
+write.table(colData(sce), opt$metadata_upd, col.names=F, row.names=FALSE, sep="\t", append=T)
```

Comment:
Actually I just checked, and the write.table isn't the issue. The problem is that the output of colData() is a DataFrame rather than a data.frame, and I think data.frame() must get called somewhere as it gets passed to write.table. We just need to short-circuit that ourselves and add the 'check.names = FALSE'. 

```suggestion
# write outputs and avoid column names mangling
write.table(data.frame(colData(sce), check.names = FALSE), opt$metadata_upd, sep="\t")
```

I'm not sure why you added the row.names = FALSE, so if you still need that please edit my suggestion.